sql-db/narayana-transactions/src/main/java/io.quarkus.ts.transactions/TransferTopUpService.java

```diff
@@ -0,0 +1,48 @@
+package io.quarkus.ts.transactions;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Named;
+
+import org.jboss.logging.Logger;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.quarkus.narayana.jta.QuarkusTransaction;
+import io.quarkus.narayana.jta.RunOptions;
+
+@ApplicationScoped
+@Named("topup")
+public class TransferTopUpService extends TransferProcessor {
+    private static final Logger LOG = Logger.getLogger(TransferTopUpService.class);
+    private final static String ANNOTATION_TOP_UP = "user top up";
+    private final static int TRANSACTION_TIMEOUT_SEC = 10;
+    private final MeterRegistry registry;
+    private long transactionsAmount;
+
+    public TransferTopUpService(MeterRegistry registry) {
+        this.registry = registry;
+        registry.gauge("transaction.topup.amount", this, TransferTopUpService::getTransactionsAmount);
```

Comment:
Do we check this somewhere?

Reply:
there is a test that are checking that the gauges are increasing and decreasing (rollback)